commands.go

```diff
@@ -68,6 +68,11 @@ var Commands = []cli.Command{
 				),
 				Value: "none",
 			},
+			cli.StringFlag{
```

Comment:
Thanks!

As you can see we are trying to move to some common operations.  There is
another PR that is doing similar things (
https://github.com/docker/machine/pull/286).  I think we are all thinking
along the same lines.  I would like to bring these together to come up with
a single "master plan" so to speak.  This will be very beneficial in the
near future, especially when we move to an API model.

As for the OS level, I'm a little torn on this.  Personally, I don't want
to try to support several distros in code.  An idea a while back was to use
cloudinit.  This would mean instead of stuff like we have here (OS
switching, deep inspection into the OS to figure out where things are, etc)
we could use cloudinit.  This way, if one wanted to use a different OS, all
they would have to do is update cloudinit (perhaps we could template it out
even).  In that way, we off-load some of the OS specific configuration to
that instead of re-creating it for machine.

+1 on refactoring the StartDocker, StopDocker, etc.  We have it for now to
support what is needed, however, I think we can plan this out for the near
future.

+1 for cli StringSlice.  +1 for prefixing errors with "Err".


On Sun, Jan 25, 2015 at 3:37 PM, Simon Thulbourn <notifications@github.com>
wrote:

> In commands.go
> <https://github.com/docker/machine/pull/38#discussion_r23507317>:
>
> > @@ -68,6 +68,11 @@ var Commands = []cli.Command{
> >  				),
> >  				Value: "none",
> >  			},
> > +			cli.StringFlag{
>
> Instead of parsing them out one by one, we can use the cli package's
> StringSliceFlag which allows you to pass the same option repeatedly and
> will return an array. Which means you could simplify hosts.go's
> parseDockerOpts() The interface to the user would be: $ docker-machine
> create -d <driver> --docker-opt="foo=bar" --docker-opt="baz=qux" ... <name>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/docker/machine/pull/38/files#r23507317>.
>